### PR TITLE
sync GBT Classifier with Spark 2.2

### PR DESCRIPTION
**Pull request by mutyonok**
- add rawPrediction and probability columns to GBT Classifier
- add loss functions for probabilistic classifiers
- move `log1pExp` to LinalgUtils
- fix for breeze Gaussian usage
- add reference config for spark 2.2
- update spark to 2.2.0-SNAPSHOT
- update tensorflow to 1.1.0

### CODE REVIEW

**Comment by mutyonok** [comment]
WIP

**Comment by hollinwilkins** [comment]
@mutyonok I want to get this PR in quickly, thanks for the work. I will have a chance to look over it on Friday after I am done giving presentations tomorrow.

**Comment by mutyonok** [comment]
@hollinwilkins no problem. there are things to discuss anyways. Waiting for your review

**Comment by hollinwilkins** [comment]
@mutyonok Looking through the PR looks fine. I want to try running it with multiple Spark versions first. Once we sort through the mllib-local dependency issue and test Spark 2.0, 2.1 and 2.2 with these changes, will merge it in.

**Comment by hollinwilkins** [comment]
@mutyonok I think a good solution for mllib-local (for now until we move to just using breeze directly) is this:
1. mleap-core depends on mllib-local 2.1
2. mllib-local is a shaded dependency of mleap-core
3. the artifact produced for mleap-core includes the shaded version of mllib-local

This means that we can develop the mleap core models/mleap runtime separately from Spark.

**Comment by hollinwilkins** [comment]
Coming back to this PR shortly. Trying to get the 0.8.0 release out shortly and I want this in there.
